src/components/Settings/RowOne.tsx

```diff
@@ -1,23 +1,119 @@
+import { useEffect, useState } from "react";
 import { TextInput } from "./TextInput";
 import { CheckMark } from "./CheckMark";
-import { AddContact } from "./AddContact";
 import { UploadButton } from "./UploadButton";
-import { EditContact } from "./EditContact";
-import { User } from "../../shared/Classes";
+import { UserDetails } from "./UserDetails";
 
 export default function RowOne({ t }: { t: Locale }) {
-  const [user, setUser] = useState(
-    new User(
-      "Anders Andersson",
-      "07012345567",
-      "anders.anderssson@företag.se",
-      "Art director"
-    )
-  );
-  const [editState, setEditState] = useState(false);
+  const router = useRouter();
+
+  // Retrieve data from database
+  const getLogos = api.exhibitor.getLogo.useQuery();
+  const getDescription = api.exhibitor.getDescription.useQuery();
+  const getJobOffers = api.exhibitor.getJobOffers.useQuery();
+
+  // Initialise mutation that are used to update database
+  const logoMutation = api.exhibitor.setLogo.useMutation();
+  const descriptionMutation = api.exhibitor.setDescription.useMutation();
+  const jobOffersMutation = api.exhibitor.setJobOffers.useMutation();
+
+  // Variables
+  const [whiteLogo, setWhiteLogo] = useState("");
+  const [colorLogo, setColorLogo] = useState("");
+  const [description, setDescription] = useState("");
+  const [checkmarks, setCheckMarks] = useState<boolean[]>([]);
+
+  function removeImageDetails(img: string) {
+    return img.replace(/^data:image\/[a-z]+\+?[a-z]+;base64,/, "");
+  }
+
+  function addImageDetails(img: string | undefined) {
+    if (!img) return "";
+    switch (img.charAt(0)) {
+      case "/":
+        return "data:image/jpg;base64," + img;
+      case "i":
+        return "data:image/png;base64," + img;
+      case "P":
+        return "data:image/svg+xml;base64," + img;
+      default:
+        return img;
+    }
+  }
+
+  function getCheckmarksPos(offer: string) {
+    let array: boolean[] = [];
+    switch (offer) {
+      case "summer":
+        array = checkmarks.slice(0, 5);
+        break;
+      case "intern":
+        array = checkmarks.slice(5, 10);
+        break;
+      case "partTime":
+        array = checkmarks.slice(10, 15);
+        break;
+    }
+    return array.reduce(
+      (out: number[], bool, index) => (bool ? out.concat(index) : out),
+      []
+    );
+  }
+
+  async function handleClick() {
+    logoMutation.mutate({
+      b64data: removeImageDetails(whiteLogo),
+      kind: "white",
+    });
+    logoMutation.mutate({
+      b64data: removeImageDetails(colorLogo),
+      kind: "color",
+    });
+    descriptionMutation.mutate(description);
+    jobOffersMutation.mutate({
+      summerJob: getCheckmarksPos("summer"),
+      internship: getCheckmarksPos("intern"),
+      partTimeJob: getCheckmarksPos("partTime"),
+      masterThesis: checkmarks[15],
+      fullTimeJob: checkmarks[16],
+      traineeProgram: checkmarks[17],
+    });
+    window.scrollTo(0, 0);
+    router.reload();
```

Comment:
If any of the above mutations fail, this reload would make all changes get lost.
Also, it isn't very pleasant to have the page reloaded. We could use `mutateAsync` and `Promise.all` here, and perhaps show a temporary message that everything was saved.

Reply:
I agree. I've added multiple temporary messages now.